scalding-core/src/main/scala/com/twitter/scalding/Stats.scala

```diff
@@ -156,3 +159,25 @@ object RuntimeStats extends java.io.Serializable {
     }
   }
 }
+
+/**
+ * FlowListener that checks counter values against a function.
+ */
+class StatsFlowListener(f: Map[StatKey, Long] => Boolean) extends FlowListener {
+
+  override def onCompleted(flow: Flow[_]): Unit = {
+    val stats = flow.getFlowStats
+    val keys = stats.getCounterGroups.asScala.flatMap(g => stats.getCountersFor(g).asScala.map(c => StatKey(c, g)))
+    val values = keys.map(k => (k, stats.getCounterValue(k.group, k.counter))).toMap
+    if (!f(values)) {
```

Comment:
if we used `f: Map[StatKey, Long] => Try[Unit]` we could just do: `f(values).get` here.